### PR TITLE
Dtsbps 462 service

### PR DESCRIPTION
**Pull request by gauravchawla73463**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSBPS-462

### Change description ###

Add search by ccd_id to /zip-files endpoint

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 1]
```diff
@@ -45,48 +46,53 @@ public ZipFileStatusService(
...
+        return Arrays.asList(new ZipFileStatus(null, null, documentControlNumber, emptyList(), emptyList()));
     }
 
     public ZipFileStatus getStatusByCcdId(String ccdId) {
         List<Envelope> envelopes = envelopeRepo.findByCcdId(ccdId);
         if (envelopes.size() > 0) {
             String zipFileName = envelopes.get(0).getZipFileName();
             List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
-            return getZipFileStatus(null, ccdId, envelopes, events);
+            return getZipFileStatus(null, ccdId,null, envelopes, events);
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:80]: formatting (space after comma)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 2]
```diff
@@ -45,48 +46,53 @@ public ZipFileStatusService(
...
         }
-        return getZipFileStatus(null, ccdId, emptyList(), emptyList());
+        return getZipFileStatus(null, ccdId,null, emptyList(), emptyList());
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:82]: formatting (space after comma)

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java [hunk 3]
```diff
@@ -15,16 +15,26 @@
...
     public final List<ZipFileEvent> events;
 
     // region constructor
-    public ZipFileStatus(String fileName, String ccdId, List<ZipFileEnvelope> envelopes, List<ZipFileEvent> events) {
+    public ZipFileStatus(
+        String fileName,
+        String ccdId,
+        String dcn,
+        List<ZipFileEnvelope> envelopes,
+        List<ZipFileEvent> events) {
```
Comment (apereverzin) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/model/out/zipfilestatus/ZipFileStatus.java:34]: formatting

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 4]
```diff
@@ -45,48 +46,53 @@ public ZipFileStatusService(
...
     }
 
     public List<ZipFileStatus> getStatusByDcn(String documentControlNumber) {
 
         List<String> zipFileNames = scannableItemRepo.findByDcn(documentControlNumber);
         List<ZipFileStatus> zipFileStatusList = new ArrayList<>();
-
-        zipFileNames.stream().forEach(
-            zipFileName ->
+        if (zipFileNames.size() > 0) {
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:57]: zipFileStatusList.isEmpty() instead of size

File: src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java [hunk 5]
```diff
@@ -45,48 +46,53 @@ public ZipFileStatusService(
...
             String zipFileName = envelopes.get(0).getZipFileName();
             List<ProcessEvent> events = eventRepo.findByZipFileName(zipFileName);
-            return getZipFileStatus(null, ccdId, envelopes, events);
+            return getZipFileStatus(null, ccdId, null, envelopes, events);
         }
-        return getZipFileStatus(null, ccdId, emptyList(), emptyList());
+        return getZipFileStatus(null, ccdId, null, emptyList(), emptyList());
     }
 
     private ZipFileStatus getZipFileStatus(
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: you can overload this method instead of adding new param

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: There will be string datatype for parameters ccdid, dcn and filename and return type is always ZipFileStatus. So overloading wouldn't be possible.

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: parameter type will always be string (either dcn, ccdid or filename) and return type will always be zipfilestatus, so overloading wouldn't be possible.

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: meant getZipFileStatusByCcdId or filename ....

Comment (gauravchawla73463) [reply src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusService.java:85]: I can do this but it will do nothing, we will have 3 functions, in the end ZipFileStatus constructor takes all of them as parameters, so we have to pass them as null anyways.

File: src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java [hunk 6]
```diff
@@ -175,8 +175,7 @@ public void should_return_list_of_ZipFileStatus_for_valid_dcn_parameter_search()
 
         // then
         assertThat(zipFileStatusList.size() == 2);
-        assertThat(zipFileStatusList.get(0).fileName).isEqualTo("hello1.zip");
-        assertThat(zipFileStatusList.get(1).fileName).isEqualTo("hello2.zip");
```
Comment (mdayican) [inline src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java:179]: why is this removed?

Comment (gauravchawla73463) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java:179]: When we are searching for dcn, filename and ccdid will be null. Out of 3 depending on what user is searching for, other two will be null. This test was written before this implementation so had to modify this.

Comment (mdayican) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java:179]: so assert them null ?

Comment (gauravchawla73463) [reply src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/services/zipfilestatus/ZipFileStatusServiceTest.java:179]: you are right. I have modified all changes to assert null which is not there.

**Review by mdayican — APPROVED** [review]

**Review by apereverzin — APPROVED** [review]
